### ISSUE TITLE
Support for Loss Functions (Non-Symmetric Loss function)

### ISSUE THREAD

**Issue by s0nicboOm** [numaproj/numalogic#73]
# Summary

Support for Non-Symmetric Loss Functions.

# Use Cases

When would you use this?

---

**Message from the maintainers**:

If you wish to see this enhancement implemented please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

**Comment by ab93** [comment]
Can you explain why we would need this @s0nicboOm ?

**Comment by s0nicboOm** [comment]
> Can you explain why we would need this @s0nicboOm ?

> Can you please explain why we would need this? @s0nicboOm

Non - Symmetric Loss function is when we do not want to weigh the positive as well as the negative outliers equally. There can be a scenario where we only want to call anomalies when the data is anomalous only in positive y-axis or only in negative y-axis.

Here there is an Opportunity of introducing more such similar loss functions.

**Comment by rum1887** [comment]
@s0nicboOm Do you want to add support for all loss functions in PyTorch?

**Comment by s0nicboOm** [comment]
> @s0nicboOm Do you want to add support for all loss functions in PyTorch?

We are ultimately looking to support loss functions that are not only in PyTorch but also provide a flexibility to the user to plug in there own custom loss function. This issue does not only point to adding more loss functions but also asks for a better way of providing that interface to the user to bring in their own custom loss function that integrates with the models that we have today.
